### PR TITLE
feat(button): add loading style; refactor tests

### PR DESCRIPTION
**Pull request by abmai**
### Button Component

Refactored tests to get more concise snapshots. Added `isLoading` style, mainly to take care of this case:

image

The spinner icon should be up to the users to add.

### CODE REVIEW

**Comment by gergelyke** [comment]
@abmai would it make sense to split it into two PRs? first, to refactor the tests, then the second to add the spinner?

File: src/button/__tests__/button.test.js [hunk 1]
```diff
@@ -24,7 +30,8 @@ describe('Button Component', () => {
     const NewStartEnhancer = () => <div />;
 
     const props = {
-      startEnhancer: 'start',
+      /* eslint-disable-next-line */
+      startEnhancer: () => <div />,
```
Comment (blackswanny) [inline src/button/__tests__/button.test.js]: i think null should work here as well as default value

**Comment by schnerd** [comment]
So I might be missing something, but does this prop actually add the spinner in? I can't find that part in the code. Or is it up to the user to still add the end enhancer? Ideally this is done automatically for them.

File: src/button/styled-components.js [hunk 2]
```diff
@@ -29,52 +38,62 @@ export const BaseButton = styled(
...
 }));
 
-function getStyleForKind({$theme, $kind}) {
+export function getStyleForShape({$theme, $shape, $size}: StylePropsT) {
+  switch ($shape) {
+    case SHAPE.round:
+    case SHAPE.square:
+      return {
+        paddingTop:
+          $theme.sizing[$size === SIZE.compact ? 'scale400' : 'scale500'],
```
Comment (blackswanny) [inline src/button/styled-components.js]: this notation is better for some IDEs as allows indexing and search better
```js
$size === SIZE.compact ? $theme.sizing.scale400 : $theme.sizing.scale500
```

**Comment by gergelyke** [comment]
Agreed with David - shouldn't this PR include the spinner too? I know we haven't implemented the component for now, but it seems misleading

**Comment by abmai** [comment]
Hmm originally I thought users might want to switch icons depending on the state of the button - i.e. I normally have a cloud icon but I want it to be a spinner when in a loading state. So it'll be more straightforward to just let the users manage their own spinner icon.

I think we can also do this by attaching our own spinner as a default `endEnhancer` if `isLoading` is true (and users can still override). But what if I want the spinner on the left, rather than on the right?

**Comment by abmai** [comment]
Going to also split this PR into 2 as requested, stay tuned!

File: src/button/styled-components.js (hunk 2 above)
Comment (abmai) [reply src/button/styled-components.js]: didn't think about that, happy to switch!

**Comment by schnerd** [comment]
> I think we can also do this by attaching our own spinner as a default endEnhancer if isLoading is true (and users can still override). But what if I want the spinner on the left, rather than on the right?

Could set endEnhancer explicitly to `null` and pass your own spinner to startEnhancer. Or we could eventually split this into two props `isLoading` and `isLoadingStyle` – `isLoadingStyle` would automatically default to the value of `isLoading`, but if you wanted to provide your own spinner you could enable `isLoadingStyle` and pass in your own enhancer.

All that said, I don't know if the "custom spinner" is a case we need to immediately handle. I'd vote to just get spinner as a default endEnhancer for now and we can revisit in the future if customers ask about it.

**Comment by abmai** [comment]
Sounds like a plan to me.

**Comment by abmai** [comment]
Closing this in favor of the new individual PRs.

**Pull request closed without merging**